hystrix/metric_collector/metric_collector.go

```diff
@@ -0,0 +1,64 @@
+package metric_collector
+
+import (
+	"sync"
+	"time"
+)
+
+var Registry = metricCollectorRegistry{
+	lock: &sync.RWMutex{},
+	registry: []func(name string) MetricCollector{
+		newDefaultMetricCollector,
+	},
+}
+
+type metricCollectorRegistry struct {
+	lock     *sync.RWMutex
+	registry []func(name string) MetricCollector
+}
+
+func (m *metricCollectorRegistry) InitializeMetricCollectors(name string) []MetricCollector {
+	m.lock.RLock()
+	defer m.lock.RUnlock()
+
+	metrics := make([]MetricCollector, len(m.registry))
+	for i, metricCollectorInitializer := range m.registry {
+		metrics[i] = metricCollectorInitializer(name)
+	}
+	return metrics
+}
+
+func (m *metricCollectorRegistry) Register(initMetricCollector func(string) MetricCollector) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
+	m.registry = append(m.registry, initMetricCollector)
```

Comment:
could be a good idea, I cant think of a specific use case off the top of my head. Should be relatively easy to do. Think we should add that api in now or I can make an issue to track it.